### PR TITLE
Caoimhe Code Review

### PR DESCRIPTION
**Pull request by caoimhegall**
In this sprint I created a new profile page that pulls data from the database. For testing, I created simple Cypress tests (found in the frontend/cypress/integration/tests folder). I made 3 commits, all the code is in Commit 2. Commit 3 is a small fix after merging with main. To run tests, run the frontend and backend and run "npx cypress open" in the frontend.

### CODE REVIEW

File: frontend/cypress/integration/tests/profilePage.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+describe('Profile page test', () => {
```
Comment (adrien-chaussabel) [inline frontend/cypress/integration/tests/profilePage.js]: Good, thorough testing

File: frontend/src/App.js [hunk 2]
```diff
@@ -29,6 +30,8 @@ class App extends Component {
           component={Registration}/>
         <PrivateRoute path='/market'
           component={VendorMarketplace}/>
+        <PrivateRoute restricted={true}
```
Comment (adrien-chaussabel) [inline frontend/src/App.js]: PrivateRoutes have no restricted field. Restricted is a Public route that shouldn't be accessible if the user is logged in.

File: frontend/src/Registration/Registration.js [hunk 3]
```diff
@@ -82,6 +82,7 @@ class Registration extends Component {
           <h2>CloudHaven Registration</h2>
         </Container>
         <TextField
+          id="username"
```
Comment (adrien-chaussabel) [inline frontend/src/Registration/Registration.js:85]: Good to add these for current & future testing

File: frontend/src/Profile/ProfileContents.js [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+        .catch((error) => {
+          console.log('error', error);
+        });
+  }
+
+  /**
+   * Render function
+   * @return {Grid}
+   */
+  render() {
```
Comment (adrien-chaussabel) [inline frontend/src/Profile/ProfileContents.js:75]: The profile component seems a bit off-center towards the right towards me

**Comment by ncpalmie** [comment]
The profile page looks great and it looks like it will be very easy to plug in the code for retrieving other user data to display like first/name once they are added to the user schema. The only thing I would suggest would be to use a more descriptive PropTypes definition in ProfileContents.js, everything else looks good to me.

File: frontend/src/Profile/ProfileContents.js [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+  }
+  /**
+   * Function needed for eslint to be happy
+   */
+  static get propTypes() {
+    return {
+      classes: PropTypes.any,
+    };
+  }
+  /**
```
Comment (ncpalmie) [inline frontend/src/Profile/ProfileContents.js]: `classes: PropTypes.shape({
        cardItems: PropTypes.style,
        cardStyle: PropTypes.style,
        large: PropTypes.style,
      })`

PropTypes.shape can be used to describe objects and 'classes' is an object containing the styles from useStyles so this type description might fit better.

File: frontend/cypress/integration/tests/register.js [hunk 6]
```diff
@@ -0,0 +1,10 @@
+describe('Register test', () => {
+  it('Registers new user', function() {
+    // need to revisit this test once the register page pushes to db
```
Comment (adrien-chaussabel) [inline frontend/cypress/integration/tests/register.js]: good to note this

**Review by adrien-chaussabel — APPROVED** [review]
Everything looks good to me besides the fact that the profile component looks off center.

**Review by jsshotts — APPROVED** [review]
Looks good!

**Review by mlkrajewski — APPROVED** [review]
Great work! Tests look thorough.

**Review by ncpalmie — APPROVED** [review]

**Pull request merged**